test/runner/e2e_delete_rel_test.cpp

```diff
@@ -19,104 +19,316 @@ class DeleteRelTest : public DBTest {
-            ->isSuccess());
-    ASSERT_TRUE(
-        conn->query(getInsertKnowsRelQuery("animal", "person", 52 /* srcID */, 0 /* dstID */))
-            ->isSuccess());
-    ASSERT_TRUE(
-        conn->query(getInsertKnowsRelQuery("animal", "person", 100 /* srcID */, 10 /* dstID */))
-            ->isSuccess());
-    // Then we delete 2 rels: animal51->person0, animal100->person10.
-    ASSERT_TRUE(
-        conn->query(getDeleteKnowsRelQuery("animal", "person", 51 /* srcID */, 0 /* dstID */))
-            ->isSuccess());
-    ASSERT_TRUE(
-        conn->query(getDeleteKnowsRelQuery("animal", "person", 100 /* srcID */, 10 /* dstID */))
-            ->isSuccess());
+    void deleteRelsAndQueryBWDList(bool isCommit, TransactionTestType transactionTestType) {
+        auto query = "MATCH (a:person)-[e:knows]->(b:animal) RETURN e.length";
+        // This check is to validate whether the query used in "deleteRelsAndQueryBWDList"
+        // scans the backward list of the knows table. If this test fails (eg. we have updated the
+        // planner), we should consider changing to a new query which scans the BWD list with
+        // the new planner.
+        validateQueryBestPlanJoinOrder(query, "HJ(a){E(a)S(b)}{S(a)}");
+        conn->beginWriteTransaction();
+        // We insert 10 rels: person[0..10]->animal0.
+        for (auto i = 0u; i < 10; i++) {
+            ASSERT_TRUE(conn->query(getInsertKnowsRelQuery("person", "animal", i, 0 /* dstID */))
+                            ->isSuccess());
+        }
+        // Then we delete rels: person[2,4,6,8]->animal0.
+        ASSERT_TRUE(
+            conn->query(getDeleteKnowsRelQuery("person", "animal", 2, 0 /* dstID */))->isSuccess());
+        ASSERT_TRUE(
+            conn->query(getDeleteKnowsRelQuery("person", "animal", 4, 0 /* dstID */))->isSuccess());
+        ASSERT_TRUE(
+            conn->query(getDeleteKnowsRelQuery("person", "animal", 6, 0 /* dstID */))->isSuccess());
+        ASSERT_TRUE(
+            conn->query(getDeleteKnowsRelQuery("person", "animal", 8, 0 /* dstID */))->isSuccess());
+        commitOrRollbackConnectionAndInitDBIfNecessary(isCommit, transactionTestType);
+        vector<string> expectedResult =
+            isCommit ? vector<string>{"0", "1", "3", "5", "7", "9"} : vector<string>{};
+        ASSERT_EQ(
+            TestHelper::convertResultToString(*conn->query(query), true /* checkOutputOrder */),
+            expectedResult);
+    }
 
-    auto actualResult = TestHelper::convertResultToString(
-        *conn->query("MATCH (:animal)-[e:knows]->(:person) return e.length"),
-        true /* checkOutputOrder */);
-    // After insertion and deletion, we are expected to see only one new rel: animal52->person0.
-    vector<string> expectedResult;
-    for (auto i = 0; i < NUM_RELS_FOR_ANIMAL_KNOWS_PERSON; i++) {
-        expectedResult.push_back(to_string(i));
+    void deleteMultipleRels(bool isCommit, TransactionTestType transactionTestType) {
```

Comment:
The previous tests above, e.g., deleteRelsFromSmall/LargList or deleteRelsFromUpdateStore  also delete multiple rels? What exactly is this testing? If this test doesn't add much, let's remove it.

Reply:
1. deleteRelsFromSmallList is to test the behaviour of deleting multiple rels from a small list. Specifically, i want to test whether the CSR structure has been fixed properly after each deletion.
2. deleteRelsFromLargelist is to test the behaviour of deleting multiple rels from a large list. Specifically, i want to test whether the largelist has been fixed properly after each deletion.
3. deleteRelsFromUpdateStore is quite different from deleting from persistent store. When we want to delete rels from updateStore, we can simply remove the rels from the insertedRelTupleIdxes of updateStore.